dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/archive/liberty/ExtensionArchiveImpl.java

```diff
@@ -61,7 +69,22 @@ public boolean isExtClassesOnly() {
     }
 
     @Override
-    public Set<String> getSPIExtensionClasses() {
-        return spiExtensions;
+    public Set<Supplier<Object>> getSPIExtensionSuppliers() {
```

Comment:
Using a Supplier seems like a good idea, so why not change the `getExtensionClasses` method too? Could be a follow up PR?

Reply:
I'll have a closer look at this. I thought there was somewhere where we needed the actual extension files, possibly because Weld was doing the instantiation based on the services files, but I'll double check that. It would be nicer to have them be consistent.

Reply:
Ideally it would be but at the moment we have some code which returns a javax Extension and other code which returns a jakarta Extension so we can't do that until we can compile against the transformed version of this bundle.